src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -252,6 +254,63 @@ private function validateAndDenormalize($currentClass, $attribute, $data, $forma
                 $data = [$data];
             }
 
+            // In XML and CSV all basic datatypes are represented as strings, it is e.g. not possible to determine,
+            // if a value is meant to be a string, float, int or a boolean value from the serialized representation.
+            // That's why we have to transform the values, if one of these non-string basic datatypes is expected.
+            //
+            // This is special to xml and csv format
+            if (
+                \is_string($data) &&
+                (XmlEncoder::FORMAT === $format || CsvEncoder::FORMAT === $format)
+            ) {
```

Comment:
It could be useful for other formats as well. IMHO this logic should be extracted in another class for re-usability, and because this class is already way too big.

Reply:
> It could be useful for other formats as well. IMHO this logic should be extracted in another class for re-usability, and because this class is already way too big.

I agree, the AbstractObjectNormalizer is getting too big and it needs a bit of refactoring. I can extract this into another class. Any ideas how to name it?

Reply:
I thought about this, sure we can put that functionality in another class. But like @dunglas already said [here](https://github.com/symfony/symfony/issues/19330#issuecomment-235606760), most of the code of the Normalizers should be extracted and used via composition rather then inheritance. 
If this is the way to go. I would rather do that in a separate PR and leave this one as it is.